Feature/home work1 2

Avaksbeorn opened the pull request:
Реализовано меню для выбора задачи. Решение каждой задачи находится в отдельном файле, названном TaskX, где X - номер задания. Задания указал в комментариях у файла задачи.

Avaksbeorn committed:
Решение 1 задачи 6 варианта: "Введите координаты точки на плоскости. Определите где она расположена (на какой оси или в каком координатном углу)."

Avaksbeorn committed:
Реализована 1 и 2 домашняя работа.

strelok372 commented on Tasks/Task1.cs:
```diff
@@ -0,0 +1,51 @@
+/*6.������� ���������� ����� �� ���������. ���������� ��� ��� ����������� (�� ����� ��� ��� � ����� ������������ ����).*/
```
Ой, а давай поправим кодировку файлов чтобы она отображалась в гитхабе?

Avaksbeorn committed:
Изменил кодировку на UTF-8

Avaksbeorn committed:
Изменил кодировку на UTF-8 и раскидал по папкам

strelok372 commented on src/Tasks/Task2.cs:
```diff
@@ -0,0 +1 @@
+﻿/*6. Составить алгоритм решения задачи: сколько можно купить  * быков, коров и телят, платя за быка 10 руб., за корову — 5 руб., а  * за теленка — 0,5 руб., если на 100 руб. надо купить 100 голов скота  */  namespace ConsoleApp1.Tasks {     public static class Task2     {         public static void Run() {              for (int bulls = 0; bulls <= 10; bulls++) {                 for (int cows = 0 ...
```
Вопрос, а ты делаешь самопроверку после того как выкладываешь пулл-реквест?

strelok372 replied:
Ещё, это слишком в лоб решение с отсутствующей гибкостью.
Если ты пишешь программу, то она должна иметь переменные, которые можно изменять, чтобы получить расчёт с другими входными данными. В ином случае программа мало чем будет отличаться от листка тетради, который исписали чернилами и потом выкинут

Avaksbeorn replied:
> Вопрос, а ты делаешь самопроверку после того как выкладываешь пулл-реквест?

Не делал, теперь буду делать.

strelok372 commented on src/Tasks/Task4.cs:
```diff
@@ -0,0 +1,83 @@
+
+            Console.WriteLine("Культуры, возделываемые на всех участках:");
+            foreach (Kult k in Enum.GetValues(typeof(Kult))) {
+                if (u1.Contains(k) && u2.Contains(k) && u3.Contains(k)) {
```
Итак, так как это задания про множества, то и использовать здесь нужно операции из алгебры множеств. Соот-но, Contains использовать нельзя)

strelok372 commented on src/Tasks/Task6.cs:
```diff
@@ -0,0 +1 @@
+﻿/*Реализовать структуру данных:   * 6.Префиксное дерево  */ namespace ConsoleApp1.Tasks {     public class TrieNode     {         public Dictionary<char, TrieNode> Children { get; set; } = new Dictionary<char, TrieNode>();         public bool IsEndOfWord { get; set; } = false;     }      public class Trie     {         private readonly TrieNode root;          public Trie() {             root = n ...
```
Не вижу тут метода для удаления слова из дерева

Avaksbeorn committed:
В задании 2 изменил программу так, чтобы она могла работать с данными пользователя (добавил только количество денег у пользователя и желаемое количество скота, как вариант можно сделать, чтобы пользователь сам вводил цену животных).
В задании 4 заменил Contains на операции из алгебры множеств
В задании 6 добавил метот для удаления слова из дерева.

strelok372 commented:
Судя по тому, что 2 и 6 задачи всё ещё в 1 строчку записаны, то я пока не проверяю?

Avaksbeorn committed:
Исправил сеперация. Проверял ее в другом репозитории, сейчас должно все работать.
